lib/spack/spack/util/s3.py

```diff
@@ -11,14 +11,23 @@
 import spack.util.url as url_util
 
 
+def get_mirror_connection(url, url_type="push"):
```

Comment:
@josephsnyder what was the thinking here? Why push_url? It's also used in `get_object` requests? Is this a bug?